### PR TITLE
SCP-4392: Migrate Nix ecosystem to use flakes.

### PR DESCRIPTION
**Pull request by shlevy**
This will require 'nix build' and 'nix develop' instead of 'nix-build' and 'nix-shell'

**PLEASE DO NOT SQUASH OR REBASE THIS PR**

### CODE REVIEW

**Comment by bwbush** [comment]
@shlevy, it seems we're generally using `cardano-node-1.35` as our base branch in order to simplify merges and linearize history. Would it be better to base this PR on that branch?

**Comment by shlevy** [comment]
@bwbush Will try once I get my tests passing

**Comment by shlevy** [comment]
@bwbush Moved to `cardano-node-1.35`

**Comment by shlevy** [comment]
This can still be reviewed, draft status is for debugging an issue with hydra/cardano-world

**Comment by shlevy** [comment]
I've confirmed the hydra issue also exists on `cardano-node-1.35`, so I'll get this in a clean mergeable state, mark ready for review, and then fix the issue as a follow-on PR

**Comment by shlevy** [comment]
Reminder: **PLEASE DO NOT SQUASH OR REBASE THIS PR**

**Comment by shlevy** [comment]
Follow-on issue, hopefully will wrap this up this weekend https://input-output.atlassian.net/browse/SCP-4402

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -55,7 +55,7 @@ jobs:
         with:
           name: ${{ env.CACHE_NAME }}
           authToken: "${{ secrets.CACHIX_AUTH_TOKEN }}"
-      - run: nix-build -A tests.nixpkgsFmt -A tests.prettier -A tests.shellcheck -A tests.stylishHaskell --arg supportedSystems '[ builtins.currentSystem ]' --restrict-eval -I .
+      - run: nix build .#tests.nixpkgsFmt .#tests.prettier .#tests.shellcheck .#tests.stylishHaskell
```
Comment (jhbertra) [inline .github/workflows/test.yml:58]: Reading these changes is great learning material

File: CONTRIBUTING.adoc [hunk 2]
```diff
@@ -62,9 +60,7 @@ bug workarounds.
 
 === How to build the code with profiling
 
-VIf you launch `nix-shell` with the `enableHaskellProfiling` argument set to true, you will get a shell where all the dependencies have been built with profiling.
-
-Like this: `nix-shell --arg enableHaskellProfiling true`.
+VIf you launch `nix develop .#prof`, you will get a shell where all the dependencies have been built with profiling.
```
Comment (jhbertra) [inline CONTRIBUTING.adoc:63]: Typo 
```suggestion
If you launch `nix develop .#prof`, you will get a shell where all the dependencies have been built with profiling.
```

File: ci.nix [hunk 3]
```diff
@@ -91,7 +86,7 @@ let
           # FIXME: this should simply be set on the main shell derivation, but this breaks
           # lorri: https://github.com/target/lorri/issues/489. In the mean time, we set it
           # only on the CI version, so that we still catch it, but lorri doesn't see it.
-          shell = (import ./shell.nix { inherit packages; }).overrideAttrs (attrs: attrs // {
+          shell = (import ./dev-shell.nix { inherit packages system; }).overrideAttrs (attrs: attrs // {
             disallowedRequisites = [ marlowe.haskell.packages.plutus-core.components.library ];
           });
```
Comment (jhbertra) [inline ci.nix:91]: With Flakes, we can likely drop Lorri support, as Lorri doesn't support Flakes anyway (And Flakes have fast cached shells and a direnv integration of their own).

File: flake.nix [hunk 4]
```diff
@@ -1,29 +1,13 @@
-# NOTE: This flake is only provided as interface to `bitte` and shouldn't be used otherwise
-#
-# Occasionally building flake builds will segfault. The workaround for this is to
-# disable the garbage collector  `GC_DONT_GC=1  nix build .#web-ghc-server
-#
-# In case you are not sure if you should be using this flake, the answer is: No.
 {
-  description = "plutus flake for pinning sources and bitte deployments";
```
Comment (jhbertra) [inline flake.nix:8]: :rofl: yeah how can you tell this was split off of a monorepo?

**Review by jhbertra — APPROVED** [review]
LGTM thanks!

File: CONTRIBUTING.adoc (hunk 2 above)
Comment (shlevy) [reply CONTRIBUTING.adoc:63]: Heh I assumed this was some formatting thing.

**Pull request merged**